### PR TITLE
Checkout preview

### PR DESCRIPTION
**Pull request by orzechdev**
It adds a checkout preview to the app.

Url to preview test app: https://master.staging.saleor.cloud/dashboard/apps/QXBwOjQw/app

Screenshot 2022-06-07 at 11 07 04

### CODE REVIEW

File: apps/checkout-app/frontend/components/templates/CustomizationDetails/CustomizationDetails.tsx
Comment (bmigirl) [inline apps/checkout-app/frontend/components/templates/CustomizationDetails/CustomizationDetails.tsx:49]: This whole part seems like strictly settings related thing, maybe it'd be worth to have it as a separate hook? Also, is there any reason this is not in `CheckoutPreviewFrame`?

**Review by bmigirl — CHANGES_REQUESTED** [review]

**Comment by bmigirl** [comment]
@orzechdev I can see here the preview url is still in env var. As we discussed, there should eventually be a text field for that. Will this happen in another pr?

**Comment by orzechdev** [comment]
> @orzechdev I can see here the preview url is still in env var. As we discussed, there should eventually be a text field for that. Will this happen in another pr?

I think it can be added in another pr as it required design update.

File: apps/checkout-app/frontend/components/templates/CustomizationDetails/CustomizationDetails.tsx
Comment (bmigirl) [reply apps/checkout-app/frontend/components/templates/CustomizationDetails/CustomizationDetails.tsx:49]: I don't think I understand. My question is why is this logic not in the CheckoutPreviewFrame component, since it's clearly passed there anyway, not used anywhere else and the logic is strictly related to it.

File: apps/checkout-app/frontend/components/elements/CheckoutPreviewFrame/CheckoutPreviewFrame.tsx
Comment (orzechdev) [reply apps/checkout-app/frontend/components/elements/CheckoutPreviewFrame/CheckoutPreviewFrame.tsx:40]: Yes, it should send settings every time they change.

File: apps/checkout-app/constants.ts
Comment (orzechdev) [reply apps/checkout-app/constants.ts:24]: This is the url of the checkout frontend. This is temporary solution. I would like to remove it in another pull request as soon as design and input to provide that value will be implemented (https://app.clickup.com/t/2549495/SALEOR-7106).

File: apps/checkout-app/constants.ts
Comment (orzechdev) [reply apps/checkout-app/constants.ts:25]: Same as preview url

File: .env
Comment (orzechdev) [reply .env:3]: ok, but how to guess app url in checkout frontend?

Comment (jwm0) [reply .env:3]: There should be some kind of input in the app, that will embed it in the iframe. We should adjust it to be more flexible, @timuric should have more info

File: apps/checkout-app/frontend/components/templates/CustomizationDetails/CustomizationDetails.tsx
Comment (orzechdev) [reply apps/checkout-app/frontend/components/templates/CustomizationDetails/CustomizationDetails.tsx:48]: I wanted to make `CheckoutPreviewFrame` component only responsible for displaying iframe with checkout, which could accept ready to send settings to that iframe in props. I've changed it to the hook returning preview seetting. Is it clear now?

File: .env
Comment (orzechdev) [reply .env:3]: Url cannot have `/api` path. Do you mean to do it like the following?
```
CHECKOUT_APP_URL=localhost:3000
CHECKOUT_API_URL=${CHECKOUT_APP_URL}/api
```

File: apps/checkout-app/constants.ts
Comment (jwm0) [reply apps/checkout-app/constants.ts:25]: I'd wait for this functionality before we merge this in, the checkout will become obsolete after it's completed and it would require a change in the vars + redeployment to make it work again

**Review by bmigirl — APPROVED** [review]

**Review by jwm0 — APPROVED** [review]

**Pull request merged**
